modules/mcuboot/Kconfig

```diff
@@ -1,5 +1,11 @@
 menu "MCUboot"
 
+config MCUBOOT_IS_PART_OF_TRUSTZONE_BUILD
```

Comment:
this is introducing a Kconfig setting with a prompt for something that is internal parent --> child image knowledge.

We could consider if we instead should exchange this information directly between CMake --> Cmake through the `child_image_preload` file.
https://github.com/nrfconnect/sdk-nrf/blob/9d20347ac77412da5c0d1dee808bf4053adc7642/cmake/multi_image.cmake#L345-L346

Note, the preload file is anyway the place where knowledge of `.../mcuboot/spm.conf` is passed, so it's easy to just pass an internal build setting instead of a file that must be parsed later.

Ref:
```
$ cat build/mcuboot/child_image_preload.cmake
...
set(OVERLAY_CONFIG .../projects/github/ncs/nrf/modules/mcuboot/spm.conf;... CACHE INTERNAL "NCS child image controlled")
```

@SebastianBoe any thoughts on this ?
The main advantage of Kconfig in this particular case is that Kconfig is well established.

Reply:
I prefer Kconfig over CMake variables for configuration of FW and pm.yml.

I want users that read FW (and other things that are configured by Kconifg, like pm.yml), to be able to understand how things became configured without reading CMake.

When we use Kconfig in this case the user will see

CONFIG_*

and know that this comes from Kconfig.

If we passed a define to pm.yml based on a CMake variable it would be less obvious where that define came from and how to control it. Than the CONFIG_* we are using now.

Reply:
Kconfig not needed anymore so removed.